File: csharp/ToolGood.Algorithm/Internals/Functions/CsharpWeb/Function_HTMLENCODE.cs
using System;
using System.Text;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.CsharpWeb
{
	internal sealed class Function_HTMLENCODE : Function_1
	{
		public Function_HTMLENCODE(FunctionBase func1) : base(func1)
		{
		}

		public override string Name => "HtmlEncode";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
		{
			var args1 = GetText_1(engine, tempParameter);
			if(args1.IsErrorOrNone) { return args1; }
			var s = args1.TextValue;
			var r = System.Web.HttpUtility.HtmlEncode(s);
			return Operand.Create(r);
		}
		public override OperandType GetResultType()
		{
			return OperandType.TEXT;
		}
	}


}

File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/internals/functions/csharpWeb/Function_HTMLENCODE.java
package toolgood.algorithm.internals.functions.csharpWeb;

import java.util.List;
import java.util.function.BiFunction;

import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Enums.OperandType;
import toolgood.algorithm.Operand;
import toolgood.algorithm.internals.NoneEngine;
import toolgood.algorithm.internals.ParameterType;
import toolgood.algorithm.internals.functions.FunctionBase;
import toolgood.algorithm.internals.functions.Function_1;

final class Function_HTMLENCODE extends Function_1 {
	public Function_HTMLENCODE(FunctionBase[] funcs) {
		super(funcs);
		if (funcs.length != 1) {
			throw new IllegalArgumentException("Function '" + Name() + "' requires exactly 1 parameter.");
		}
	}

	@Override
	public String Name() {
		return "HtmlEncode";
	}

	@Override
	public Operand Evaluate(AlgorithmEngine engine, BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
		Operand args1 = GetText_1(engine, tempParameter);
		if (args1.IsErrorOrNone()) {
			return args1;
		}
		String s = args1.TextValue();
		String r = HtmlEncode(s);
		return Operand.Create(r);
	}

	private static String HtmlEncode(String s) {
		if (s == null || s.isEmpty()) {
			return s;
		}
		return s.replace("&", "&amp;")
				.replace("<", "&lt;")
				.replace(">", "&gt;")
				.replace("\"", "&quot;")
				.replace("'", "&#39;");
	}

	@Override
	public OperandType GetResultType() {
		return OperandType.TEXT;
	}

	@Override
	void GetParameterTypes(NoneEngine noneEngine, List<ParameterType> result, OperandType operandType, String op, String val) {
		func1.GetParameterTypes(noneEngine, result, OperandType.TEXT);
	}
}
